gui/src/app/SPAnalysis/SetupSPAnalysis.tsx

```diff
@@ -0,0 +1,77 @@
+import { FunctionComponent, PropsWithChildren, useMemo, useReducer } from "react"
+import { SPAnalysisContext } from "./SPAnalysisContext"
+
+type SetupSPAnalysisProps = {
+    sourceDataUri: string
+}
+
+type AnalysisFiles = {
+    [key: string]: string
+}
+
+type AnalysisFilesAction = {
+    type: 'set'
+    key: string
+    value: string
+} | {
+    type: 'delete'
+    key: string
+}
+
+const analysisFilesReducer = (state: AnalysisFiles, action: AnalysisFilesAction): AnalysisFiles => {
+    switch (action.type) {
+        case 'set': {
+            return {
+                ...state,
+                [action.key]: action.value
+            }
+        }
+        case 'delete': {
+            const newState = { ...state }
+            delete newState[action.key]
+            return newState
+        }
+    }
+}
+
+const SetupSPAnalysis: FunctionComponent<PropsWithChildren<SetupSPAnalysisProps>> = ({ children }) => {
+    const [analysisFiles, dispatchAnalysisFiles] = useReducer(analysisFilesReducer, {})
+    const value = useMemo(() => {
+        return {
+            localDataModel: {
+                title: 'SPAnalysis',
```

Comment:
I have changed this to 'Untitled' and added the comment:

"// title is hard-coded for now because we don't yet have a mechanism for it to be changed"